### PR TITLE
replace eth package level config with object/func options

### PR DESCRIPTION
**Pull request by xiaozhou**
- replace eth package level config with object/func options
- fix forever blocking if waitmined errored

### CODE REVIEW

File: eth/options.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import (
+	"math/big"
+	"time"
+)
+
+type txOptions struct {
+	ethValue           *big.Int      // transact
+	minGasGwei         uint64        // transact
+	maxGasGwei         uint64        // transact
```
Comment (i9) [inline eth/options.go:13]: nit: consider add a field plusGasGwei meaning add this much on top of recommended gas.
and I don't think maxGasGwei ever helps, instead we have to constantly update rtconfig due to congested mainnet.
or if we say this is to safe guard any mistake setting ridiculous gas price, then maxGasGwei should be really large like 100 or even higher

Comment (xiaozhou) [reply eth/options.go:13]: Added the new field. Agree about maxGasGwei, we should keep this and set to a very large value in rtconfig.

File: eth/options.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+const (
+	defaultPollingInterval      = 15 * time.Second
+	defaultTxTimeout            = 6 * time.Hour
+	defaultTxQueryTimeout       = 2 * time.Minute
+	defaultTxQueryRetryInterval = 10 * time.Second
+)
+
+func defaultTxOptions() txOptions {
```
Comment (rdcel) [inline eth/options.go:30]: nit: It's more "go" to return "*txOptions" and it seems only its pointer is used below

Comment (xiaozhou) [reply eth/options.go:30]: I was using grpc-go as a reference: https://github.com/grpc/grpc-go/blob/v1.29.1/dialoptions.go#L562

One thing special there is that we always use deep copy when using the default options in our code, e.g., https://github.com/celer-network/goutils/blob/f9b58745ee/eth/transactor.go#L116-L119

Comment (rdcel) [reply eth/options.go:30]: Ah, now I see as I read more code, it's always modifying a new fresh code.  Makes sense. thanks.

Comment (xiaozhou) [reply eth/options.go:31]: done

**Pull request merged**